database/database.go

```diff
@@ -24,7 +24,11 @@ type Database interface {
 
 	// SaveTx will be called to save each transaction contained inside a block.
 	// An error is returned if the operation fails.
-	SaveTx(tx *types.Tx) error
+	SaveTx(tx *types.Tx, partition_id int64) error
+
+	// CreateTxPartition create tx postgres partition table if not already exist.
+	// An error is returned if the operation fails.
+	CreatePartition(table string, height int64) (int64, error)
```

Comment:
I think this method should be moved inside the `Database` interface implementation that uses PostgreSQL underneath. If we put it here inside the generic `Database` interface, all databases types need to implement it, which might not always be possible (eg. inside MongoDB there is not even the concept of partition as far as I know)